impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Gauge.js

```diff
@@ -0,0 +1,155 @@
+            formatter: "{value}"
+          },
+          itemStyle: {
+            borderColor: "#fff",
+            borderWidth: 1
+          },
+          emphasis: {
+            label: {
+              fontSize: 20
+            }
+          }
+        }
+      ];
+    },
+
+    _buildData: function() {
+      var dataTable = this.model.data;
+
+      if(dataTable.originalCrossTable) {
+        dataTable = dataTable.originalCrossTable.toPlainTable({skipRowsWithAllNullMeasures: true});
+      }
+
+      var rowLength = dataTable.getNumberOfRows();
+      var colLength = dataTable.getNumberOfColumns();
+      var font = util.getDefaultFont(null, 12);
+
+      var records = [];
+
+      var maxWidth = this.model.width;
+      var midIdx = Math.floor(rowLength / 2); // Calculate the index of the middle element
+
+      var initX = 0; // Midpoint x-coordinate
+      var yPercentage = 100; // Constant y-coordinate in percentage
+      var separation = (maxWidth / (rowLength - 1)) * 0.8; // Dynamic separation between elements
+
+      // Read data from dataTable and push it to records
+      for(var i = 0; i < rowLength; i++) {
+        var offsetX;
+
+        if(rowLength % 2 === 0) { // If even number of data elements
+          if(i < midIdx) { // Elements before the middle
+            offsetX = -((midIdx - 0.5 - i) * separation);
+          } else { // Elements after the middle
+            offsetX = (i - midIdx + 0.5) * separation;
+          }
+        } else { // If odd number of data elements
+          if(i === midIdx) { // Middle element
+            offsetX = 0;
+          } else if(i < midIdx) { // Elements before the middle
+            offsetX = -((midIdx - i) * separation);
+          } else { // Elements after the middle
+            offsetX = (i - midIdx) * separation;
+          }
+        }
+
+        records.push({
+          name: dataTable.getFormattedValue(i, colLength - 2),
+          value: dataTable.getValue(i, colLength - 1),
+          tooltip: this._buildTooltip(this._buildRowTooltipHtml(dataTable, i), font),
+          title: {
+            offsetCenter: [initX + offsetX, yPercentage + "%"],
+            overflow: "truncate",
+            width: 100
+          },
+          detail: {
+            offsetCenter: [initX + offsetX, (yPercentage + 15) + "%"]
+          }
+        });
+      }
+
+      return records;
+    },
+
+    /** This method initializes the eChart library,
+     *configures all the properties, data that are required for the chart and renders it.
+     *
+     * @override */
+    _updateAll: function() {
+
+      this.base();
+
+      var myChart = this._initializeChart();
+      var records = this._buildData();
+      var options = {
+        tooltip: {
+          trigger: "item"
+        },
+        color: this._buildColors(),
+        series: this._buildSeries()
+      };
+
+      this._configureData(options, records);
+
+      // Draw the chart
+      myChart.setOption(options);
+    }
+  })
```

Comment:
Is there anything specific about this implementation that could not be moved to the base class?
See other related comment regarding the `_configureLabel` move to the base class.